lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -1031,7 +1060,11 @@ SILDeclRef SILDeclRef::getNextOverriddenVTableEntry() const {
     // Distributed thunks are not in the vtable.
     if (isDistributedThunk())
       return SILDeclRef();
-    
+
+    // Back deployment support functions are not in the vtable.
+    if (backDeploymentKind != SILDeclRef::BackDeploymentKind::None)
```

Comment:
Could this be instead enforced with a restriction preventing @_backDeploy from being applied to non-final methods?

Reply:
Yeah, if something in SILGen is impossible and should have been caught by the type checker, asserting is the right approach. Trust but verify :)

Reply:
> So it turns out that this is called unconditionally and the early return is needed.

Ah, this needs more thought then. When it is being called? I see that we call this for final methods that override something else. So the question is, is this supported:

```swift
class Base {
  func foo() {}
}

class Derived {
  @_backDeploy final override func foo() {}
}
```

If so, this should do something more useful since back-deployed methods _can_ appear in the vtable, they just cannot be overriden further.

If not, you might want to prohibit non-final *and* override methods from being @_backDeploy.

Reply:
Sorry, I goofed here and put the assert before of the `if (auto overridden = getOverridden())` branch which caused it to fail. It's fine to assert inside the branch so I've done that.